### ISSUE TITLE
Comparison YOLOv8 vs. YOLOv5: worse detection after training with custom dataset

### ISSUE THREAD

**Issue by RobbStarkAustria** [ultralytics/ultralytics#734]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training, Validation, Detection

### Bug

I have done a comparison with the same dataset on both, YOLOv8 and YOLOv5. I used the same dataset and the same hyperparameters. Training was stopped by early stopping patience 20. Image-size is 1024x1024 which was previously rescaled.

In the dataset are some similar objects to detect like: (for example, cutted from the original much bigger image)
![grafik](https://user-images.githubusercontent.com/7694336/215702096-3b2add32-1357-42cb-9bd1-454d250ffb6c.png) ![grafik](https://user-images.githubusercontent.com/7694336/215702281-861c56a0-83eb-43ba-92d7-7167553b778d.png)

The detection results are this;
YOLOv5:
![grafik](https://user-images.githubusercontent.com/7694336/215703164-2cab9edb-964b-4714-80e9-1a6df74f8d5b.png) ![grafik](https://user-images.githubusercontent.com/7694336/215703231-5c765d38-eb06-4b44-8435-0122919729aa.png)
fine detection

YOLOv8:
![grafik](https://user-images.githubusercontent.com/7694336/215705456-fc658788-abac-4674-bb63-1f1595bc7fd3.png) ![grafik](https://user-images.githubusercontent.com/7694336/215705583-157e0769-9f8c-48a8-b7bd-4edc5e89876a.png)
As you see the right image is detected correct, but the left image has a double detection with nearly the same bounding box. The left image is only an example for a lot of double detected ROI on the image. The probabilities are often very similar for both detections.

I'm not sure if it is a detection, validation or training problem. I will give you the outputs from both experiments:
**Validation YOLOv5**

    Model summary: 322 layers, 86233975 parameters, 0 gradients, 203.9 GFLOPs
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|██████████| 96/96 [00:31<00:00,  3.02it/s]
                   all        383      44622       0.99      0.992      0.993      0.765
                  fu-d        383       2758      0.996      0.997      0.994      0.811
                  fu-u        383       1851      0.994      0.995      0.993      0.786
                sb-min        383        112      0.964      0.973      0.988      0.662
                  mi-d        383       9725      0.998      0.997      0.994       0.79
                  mi-u        383       7367      0.999      0.999      0.995      0.761
                 sebre        383       6134      0.998      0.997      0.995      0.702
                  sm-d        383       9932      0.998      0.996      0.994      0.781
                  sm-u        383       6461      0.998      0.996      0.995       0.78
                  sf-d        383        156      0.971      0.987      0.993      0.791
                  sf-u        383        126      0.984      0.983       0.99      0.787

**Validation YOLOv8**
```
 val: Scanning /home/olaf/PycharmProjects/datasets/note_sheets/labels/val.cache... 383 images, 30 backgrounds, 0 corrupt: 100%|██████████| 383/383 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 96/96 [00:42<00:00,  2.26it/s]
                   all        383      44622      0.995      0.995      0.993      0.792
                  fu-d        383       2758      0.998          1      0.994      0.836
                  fu-u        383       1851      0.998      0.998      0.994      0.825
                sb-min        383        112      0.975      0.973      0.985      0.677
                  mi-d        383       9725      0.999      0.998      0.995      0.817
                  mi-u        383       7367      0.999      0.999      0.995      0.796
                 sebre        383       6134      0.999      0.998      0.995      0.729
                  sm-d        383       9932      0.998      0.998      0.995      0.807
                  sm-u        383       6461      0.998      0.998      0.995      0.811
                  sf-d        383        156      0.992      0.994      0.995      0.809
                  sf-u        383        126      0.991      0.992      0.992      0.809
```

**Result-plot YOLOv5**
![grafik](https://user-images.githubusercontent.com/7694336/215710457-e6b97e2b-3940-4e8c-8883-a0d56b86c7ab.png)

**Result-plot YOLOv8**
![grafik](https://user-images.githubusercontent.com/7694336/215710313-d196d580-c6b0-4896-8555-9d60d6bfc9fa.png)

Comparing the two outputs, I find it striking that the starting values and the increases of the metrics are so different. Especially in the YOLOv8 experiment, the first value for the mAP_0.5 and mAP_0.5:095 seems too high to me compared to YOLOv5.

So much for my observations. I am, as I said, not sure if the problem is in the training, validation or prediction domain. 
I would like to ask you to investigate these behaviours. If you need more data, please feel free to contact me.

Kind regards

### Environment

YOLOv5 🚀 v8.0.22
Python-3.10.8 torch-1.13.1
CUDA:0 (NVIDIA GeForce RTX 3070, 8192MiB)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by bit-scientist** [comment]
can you also share the confidence scores of those detected bboxes for both v5 and v8 models? Please share your training commands for both models.

**Comment by RobbStarkAustria** [comment]
**Below the desired details:** (hope it helps)
confidences:
YOLOv5:
left image: 0.848961
right image: 0.829423

YOLOv8:
left image: (green box) 0.863456, (orange box) 0.800002
right image: 0.853854

**Training parameters YOLOv5:**
```
        epochs=150,
        weights="yolov5x.pt",
        batch_size=2,
        device=0,
        hyp="hyp.my-augmentation.yaml",
        imgsz=1024,
        optimizer="Adam",
        lr0=0.001,
        cos_lr=True,
        patience=20,
```
**hyperparameter YOLOv5**
```
lr0: 0.001  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.1  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 3.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.1  # warmup initial bias lr
box: 0.05  # box loss gain
cls: 0.3  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 0.7  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
# anchors: 3  # anchors per output layer (0 to ignore)
# this parameters are all zero since we want to use albumentation framework
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 0.0  # image rotation (+/- deg)
translate: 0.1  # image translation (+/- fraction)
scale: 0  # image scale (+/- gain)
shear: 0  # image shear (+/- deg)
perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.0  # image flip left-right (probability)
mosaic: 1.0  # image mosaic (probability)
mixup: 0.0  # image mixup (probability)
copy_paste: 0.0  # segment copy-paste (probability)
```
**Training parameters YOLOv8:**
```
task: detect
mode: train
model: yolov8x.yaml
data: ./data/diamant_notes.yaml
epochs: 150
patience: 20
batch: 2
imgsz: 1024
save: true
cache: false
device: 0
workers: 8
project: diamant_notes
name: max_epochs
exist_ok: false
pretrained: false
optimizer: Adam
verbose: true
seed: 0
deterministic: true
single_cls: false
image_weights: false
rect: false
cos_lr: true
close_mosaic: 10
resume: false
overlap_mask: true
mask_ratio: 4
dropout: false
val: true
save_json: false
save_hybrid: false
conf: 0.001
iou: 0.7
max_det: 1000
half: false
dnn: false
plots: true
source: ultralytics/assets/
show: false
save_txt: false
save_conf: false
save_crop: false
hide_labels: false
hide_conf: false
vid_stride: 1
line_thickness: 3
visualize: false
augment: true
agnostic_nms: false
classes: null
retina_masks: false
boxes: true
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: 17
workspace: 4
nms: false
lr0: 0.001
lrf: 0.01
momentum: 0.937
weight_decay: 0.0010078125
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
fl_gamma: 0.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.0
shear: 0
perspective: 0.0
flipud: 0.0
fliplr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
cfg: null
v5loader: false
save_dir: diamant_notes/max_epochs

```

**Comment by Laughing-q** [comment]
@RobbStarkAustria hi could you provide more data? I'd like to reproduce and investigate it. Thanks!
here's my email: laughingq@163.com. you can send me the data if you don't mind. :)

**Comment by RobbStarkAustria** [comment]
@Laughing-q I will sent you some data. Details in the mail

**Comment by AidanAbramson** [comment]
@RobbStarkAustria I am also getting double BBoxes on onjects after training on custom data. My solution was to force NMS as a postprocess. Lmk if you found a better solution.

**Comment by RobbStarkAustria** [comment]
@AidanAbramson I have tried it, NMS is working, but there are still double detections. Maybe it is because of the high number of bboxes on a image to predict. @Laughing-q will take a look on this behavior.

**Comment by Laughing-q** [comment]
@RobbStarkAustria hi, I noticed you used `augment=True` for training, but actually this `augment` arg is for predicting. [https://github.com/ultralytics/ultralytics/blob/21ae321bc2744a334788d4c3f617fb0698950b31/ultralytics/yolo/engine/predictor.py#L165](https://github.com/ultralytics/ultralytics/blob/21ae321bc2744a334788d4c3f617fb0698950b31/ultralytics/yolo/engine/predictor.py#L165)
You're getting overlapped boxes because of `augment=True`. Please set `augment=False` for predicting.

**Comment by RobbStarkAustria** [comment]
Many thanks for the solution to the problem and the extensive discussion via mail. You are great!

I will avoid augment=True for further trainings.

**Comment by glenn-jocher** [comment]
@RobbStarkAustria we've removed `augment` from being remembered starting in 8.0.29 for release today, so this issue should be resolved now.
